src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -0,0 +1,254 @@
+    /// if the file exists in the current directory, else an empty string.
+    /// </summary>
+    /// <param name="environmentValue">Name of the environment to
+    /// generate the config file name for.</param>
+    /// <param name="considerOverrides">whether to look for overrides file or not.</param>
+    /// <returns></returns>
+    public string GetFileName(string? environmentValue, bool considerOverrides)
+    {
+        string fileNameWithoutExtension = _fileSystem.Path.GetFileNameWithoutExtension(_baseConfigFileName);
+        string fileExtension = _fileSystem.Path.GetExtension(_baseConfigFileName);
+        string configFileName =
+            !string.IsNullOrEmpty(environmentValue)
+            ? $"{fileNameWithoutExtension}.{environmentValue}"
+            : $"{fileNameWithoutExtension}";
+        string configFileNameWithExtension = $"{configFileName}{fileExtension}";
+        string overriddenConfigFileNameWithExtension = GetOverriddenName(configFileName);
+
+        if (considerOverrides && DoesFileExistInCurrentDirectory(overriddenConfigFileNameWithExtension))
+        {
+            return overriddenConfigFileNameWithExtension;
+        }
+
+        if (DoesFileExistInCurrentDirectory(configFileNameWithExtension))
+        {
+            return configFileNameWithExtension;
+        }
+
+        return string.Empty;
+    }
+
+    private static string GetOverriddenName(string fileName)
+    {
+        return $"{fileName}.overrides{CONFIG_EXTENSION}";
+    }
+
+    /// <summary>
+    /// Generates the name of the file based on environment value.
+    /// NOTE: Input File name should not contain extension
+    /// </summary>
+    public static string GetEnvironmentFileName(string fileName, string environmentValue)
+    {
+        return $"{fileName}.{environmentValue}{CONFIG_EXTENSION}";
+    }
+
+    public bool DoesFileExistInCurrentDirectory(string fileName)
+    {
+        string currentDir = _fileSystem.Directory.GetCurrentDirectory();
+        // Unable to use ILogger because this code is invoked before LoggerFactory
+        // is instantiated.
+        if (_fileSystem.File.Exists(_fileSystem.Path.Combine(currentDir, fileName)))
+        {
+            // This config file is logged as being found, but may not actually be used!
+            Console.WriteLine($"Found config file: {fileName}.");
+            return true;
+        }
+        else
+        {
+            // Unable to use ILogger because this code is invoked before LoggerFactory
+            // is instantiated.
+            Console.WriteLine($"Unable to find config file: {fileName} does not exist.");
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// This method reads the dab.draft.schema.json which contains the link for online published
+    /// schema for dab, based on the version of dab being used to generate the runtime config.
+    /// </summary>
+    public string GetPublishedDraftSchemaLink()
```

Comment:
Not needed in this PR, but we'll probably want to move this as an abstract method on the base class, as other implementations are likely to need to get the schema link. The implementation _might_ also be valid in the base class, but we can evaluate that when the time comes.

Reply:
Makes sense. i will move it as part of this pr as this seems like a very valid thing to have in the base class.